### ISSUE TITLE
Permission to access certain folders on macOS

### ISSUE THREAD

**Issue by pck1980** [microsoft/vscode#112647]
- VSCode Version: 1.53.0-exploration
- OS Version: macOS Big Sur 11.1

Steps to Reproduce:

1. Try to save a file to Desktop or Documents or Downloads folder
2. Do not grant the permission you are asked for
3. The file gets saved there nonetheless

Detailed description:
I am not sure if this is a bug of macOS, VS Code, or both. I noticed that when trying to access one of these folders that are "protected" by macOS privacy settings, the usual dialog appears to ask whether to allow the application to access the folder or not. But no matter what you choose, access seems to be possible afterwards.
"Code - Exploration" is listed under Privacy -> Files and Folders afterwards, but not matter whether the checkboxes are checked or not, access is possible.
I don't understand this "feature" of macOS, as there are apps that are able to access these folders without ever asking (and which also do not appear in the list under Privacy -> Files and Folders), and there are others (like Terminal.app) which ask and seem to "respect" the choice that was made, but I think asking and then ignoring the user's choice is worse than not asking at all ;-)

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by bpasero** [comment]
I don't think there is a change in VSCode we can or should make that has an impact. I suggest to file a bug against Apple.

**Comment by pck1980** [comment]
I also sent a security report to Apple, as I also expected this to be a bug of macOS.
However, as there are programs that don't ask for or need this permission to access these folders at all, it somehow also seems to depend on the program, though I have no idea how.

**Comment by pck1980** [comment]
I just tried a few more apps and it seems you're right and the problem is not specific to VSCode.
It seems that most (but not all) apps downloaded from the App Store don't ask for that permission at all, and are simply able to access those folders.
Apps not downloaded from the App Store show the dialog, but for most of them, it seems not to matter which choice you make, and the folders are accessible afterwards. The only exceptions I found so far are Terminal.app and Blender.
So it's probably a bug Apple has to fix.
I think this issue can be closed.
